lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -380,15 +386,46 @@ def max_compact_name_size
       50
     end
 
+    def inspect_with_allocation_control(slice)
+      x = Thread.current
+      
+      start_alloc_size = ObjectSpace.memsize_of_all
+      
+      trace = TracePoint.new(:c_call, :call) do |tp|
+        curr_alloc_size = ObjectSpace.memsize_of_all
+        
+        if(curr_alloc_size - start_alloc_size > 1e7)
+          
+          trace.disable
+          x.raise StandardError, "Out of memory: evaluation took longer than 10mb." if x.alive?
```

Comment:
Please create new error class (and modify `rescue`) to be able to catch expected errors, not all of them.